File: Libraries/Json/resources/python/json.py

def lib_json_parse(g, raw):
  import json
  try:
    return lib_json_parse_json_thing(g, json.loads(raw))
  except:
    return None

def lib_json_parse_json_thing(g, item):
  if item == None: return v_buildNull(g)
  t = str(type(item))
  if "'bool'" in t:
    return v_buildBoolean(g, item)
  if "'int'" in t or "'long'" in t:
    return v_buildInteger(g, item)
  if "'float'" in t:
    return v_buildFloat(g, item)
  if "'string'" in t or "'unicode'" in t:
    return v_buildString(g, str(item))
  if "'list'" in t:
    output = []
    for o in item:
      output.append(lib_json_parse_json_thing(g, o))
    return v_buildList(output)
  if "'dict'" in t:
    keys = []
    values = []
    for key in item.keys():
      keys.append(key)
      values.append(lib_json_parse_json_thing(g, item[key]))
    return v_buildDictionary(keys, values)
  return v_buildNull(g);

File: Libraries/Json/resources/javascript/json.js

LIB$json$parseJson = function(globals, rawText) {
    try {
        return LIB$json$convertJsonThing(globals, window.JSON.parse(rawText));
    } catch (e) {
        return null;
    }
};

LIB$json$convertJsonThing = function(globals, thing) {
    var type = C$common$typeClassify(thing);
    switch (type) {
        case 'null': return v_buildNull(globals);
        case 'bool': return v_buildBoolean(globals, thing);
        case 'string': return v_buildString(globals, thing);
        case 'list':
            var list = [];
            for (i = 0; i < thing.length; ++i) {
                list.push(LIB$json$convertJsonThing(globals, thing[i]));
            }
            return v_buildList(list);
        case 'dict':
            var keys = [];
            var values = [];
            for (var rawKey in thing) {
                keys.push(rawKey);
                values.push(LIB$json$convertJsonThing(globals, thing[rawKey]));
            }
            return v_buildStringDictionary(globals, keys, values);
        case 'int':
            return v_buildInteger(globals, thing);
        case 'float':
            return v_buildFloat(globals, thing);
        default:
            return v_buildNull(globals);
    }
};
